bootstore/src/schemes/v0/peer.rs

```diff
@@ -765,9 +777,12 @@ impl Node {
                     return;
                 };
 
-                // Ignore the stale message if the unique_id doesn't match what
-                // we have stored.
+                // Put back the non-matching connection we removed
+                // The received message is stale, so we return.
                 if unique_id != accepted_handle.unique_id {
+                    self.accepted_connections
```

Comment:
What does it mean to put this stale connection back on the accepted_connections list?

Reply:
TLDR: The incoming message is stale, but the connection we put back is always current. Otherwise it wouldn't be in the map.

We are actually putting back the original connection. `unique_id` here is from the received message on line 757 and essentially says:  "This message come from this tokio task". However, we remove the handle matching the the address in the message on line 765 *before* we do the check. `accepted_handle` holds the current unique_id. When we do the check on line 782 we realize that the message is actually not current, and so we don't want to progress the handle from `self.accepted_connections` into `self.established_connections`. But we don't want to discard our current handle just for a stale message, so we put it back where it was in `self.accepted_connections`. 

I could have instead borrowed to do the check and removed after the borrow as done elsewhere. Maybe that would have been clearer. All in all, I think this code is more confusing than it has to be and I plan to add some more tests and clean it up very soon.